### PR TITLE
BUG: Renewable Controller edge case

### PR DESCRIPTION
**Pull request by lukelowry**
## Description

Improves the numerical conditioning of the REGCA and REECB algebraic equations and cleans up their documentation and tests.

## Proposed changes

- Replace REGCA's implicit HVRCM constraint with an explicit smooth gain law.
- Remove REECB's `ILMAX` continuation variable and compute current-circle capacity directly.
- Update initialization and documentation to match the revised equations.
- Replace smoothing-specific answer keys with ideal values and shared smoothing tolerances while retaining Enzyme and dependency-tracking coverage.

## Checklist

- [x] All tests pass.
- [x] Code compiles cleanly with flags `-Wall -Wpedantic -Wconversion -Wextra`.
- [x] The new code follows GridKit™ style guidelines.
- [x] There are unit tests for the new code.
- [x] The new code is documented.
- [x] The feature branch is rebased with respect to the target branch.
- [x] Changelog changes are N/A; this is a focused cleanup of existing model implementations.

## Further Comments

This was contributing to some convergence failures when a particularly low-impedance fault is applied.

### CODE REVIEW

**Comment by lukelowry** [comment]
I think there is an inconsistant default tolerance issue that is causing issues across platforms because the only reason I observed this issue is that the CI for #564 is failing, but works on my machine

File: tests/UnitTests/PhasorDynamics/ControllerReecbTests.hpp [hunk 1]
```diff
@@ -819,14 +835,27 @@ namespace GridKit
         {
           if (index(expected_residuals[row].variable) != row)
           {
-            std::cout << "REECB residual key position " << row << " names row "
+            std::cout << "REECB residual expectation position " << row << " names row "
```
Comment (nkoukpaizan) [inline tests/UnitTests/PhasorDynamics/ControllerReecbTests.hpp]: We'll need to use the logger for these, moving forward.

File: tests/UnitTests/PhasorDynamics/ConverterRegcaTests.hpp [hunk 2]
```diff
@@ -165,34 +172,32 @@ namespace GridKit
...
-                                 0.60000000000018872,
-                                 "initialized IP");
-        success         *= scalarMatches(latched_y[index(Vars::IQ)], 0.2, "initialized IQ");
+        success         *= scalarMatches(latched_y[index(Vars::PBR)], 0.6, "latched PBR");
+        success         *= scalarMatches(latched_y[index(Vars::QBR)], 0.2, "latched QBR");
 
-        latched_y[index(Vars::IP)] = 0.5;
-        latched_y[index(Vars::IQ)] = 0.14;
+        latched_y[index(Vars::IP)] -= 0.1;
+        latched_y[index(Vars::IQ)] -= 0.06;
```
Comment (nkoukpaizan) [inline tests/UnitTests/PhasorDynamics/ConverterRegcaTests.hpp]: Looking only at the diff, it was not obvious that these were arbitrary, but I see the comment above. I upcoming cleanup, I recommend making these things clearer. E.g., 
```suggestion
        latched_y[index(Vars::IP)] -= 0.1; // arbitrary current displacement
        latched_y[index(Vars::IQ)] -= 0.06; // arbitrary current displacement
```

File: tests/UnitTests/PhasorDynamics/ControllerReecbTests.hpp [hunk 3]
```diff
@@ -1661,29 +1712,119 @@ namespace GridKit
...
+          {
+            setJacobianState(transition, 0.1, 0.2, ipcmd);
+            if (transition.evaluate() != 0 || transition.reecb.evaluateJacobian() != 0)
+            {
+              std::cout << "REECB Jacobian transition failed for " << label << '\n';
+              success = false;
+              return {};
+            }
+
+            auto* coo = transition.reecb.getCooJacobian();
```
Comment (nkoukpaizan) [inline tests/UnitTests/PhasorDynamics/ControllerReecbTests.hpp]: I would use `getCsrJacobian()` and the `MapFromCsr` utility to convert to `DependencyMap`. I see there is a similar pattern in `GovernorGastPtiTests`. `CooJacobian` was only meant as a construction helper in the model implementation.

**Review by nkoukpaizan — APPROVED** [review]
A few minor comments.

File: tests/UnitTests/PhasorDynamics/ControllerReecbTests.hpp (hunk 3 above)
Comment (lukelowry) [reply tests/UnitTests/PhasorDynamics/ControllerReecbTests.hpp]: I changed it in [dad62ed](https://github.com/ORNL/GridKit/pull/568/commits/dad62ed3c5dff600445d7adbc1733e0ccbc755f0) and its cleaner now. let me know what you think

File: tests/UnitTests/PhasorDynamics/ControllerReecbTests.hpp (hunk 1 above)
Comment (lukelowry) [reply tests/UnitTests/PhasorDynamics/ControllerReecbTests.hpp]: I changed it to logger in [dad62ed](https://github.com/ORNL/GridKit/pull/568/commits/dad62ed3c5dff600445d7adbc1733e0ccbc755f0), thank you
